crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -707,6 +707,18 @@ private void checkClass(Class<?> crdClass) {
         } else {
             for (Class c : subtypes(crdClass)) {
                 hasAnyGetterAndAnySetter(c);
+                try {
+                    String typePropertyName = crdClass.getAnnotation(JsonTypeInfo.class).property();
+                    String methodName = "get" + typePropertyName.substring(0, 1).toUpperCase(Locale.ENGLISH) + typePropertyName.substring(1).toLowerCase(Locale.ENGLISH);
+                    @SuppressWarnings("unchecked")
+                    Method method = c.getMethod(methodName);
+
+                    if (!hasAnnotatedType(method)) {
+                        err(c.getCanonicalName() + "#" + methodName + " is not annotated with @JsonInclude(JsonInclude.Include.NON_NULL)");
+                    }
+                } catch (NoSuchMethodException e) {
+                    err(e.getMessage());
+                }
```

Comment:
This could be factored into a method called something like `checkDiscriminatorIsIncluded`